h2o-core/src/main/java/hex/grid/HyperSpaceWalker.java

```diff
@@ -222,14 +234,28 @@ public MP getParams() {
     public ModelParametersBuilderFactory<MP> getParametersBuilderFactory() {
       return _paramsBuilderFactory;
     }
-
-    protected MP getModelParams(MP params, Object[] hyperParams) {
+    
+    private Map<String, Object[]>[] extractSubspaces() {
+      if(_hyperParams.get(SUBSPACES) == null) { return new Map[0]; }
+      Map<String, Object[]>[] subspaces = new Map[_hyperParams.get(SUBSPACES).length];
+      for (int i = 0; i < _hyperParams.get(SUBSPACES).length; i++) {
+        subspaces[i] = (Map<String, Object[]>)_hyperParams.get(SUBSPACES)[i];
+      }
```

Comment:
not sure you still need to do that now that you ensured that subspaces value is of the correct type.
I think
```java
return (Map<String, Object[]>[])_hyperParams.get(SUBSPACES);
```
may work just fine.

Reply:
Casting directly seemed to throw an error, so I used
```
return Stream.of(_hyperParams.get(SUBSPACES)).toArray(Map[]::new);
```
If you think there's a better way to do this, let me know.